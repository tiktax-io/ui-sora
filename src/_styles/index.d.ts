/******************************************************************************
 * Below are all global rules applied on theme by default.
 *****************************************************************************/
export declare const BREAKPOINTS: Array<number>
export declare const TYPOGRAPHY_BASE: number
export declare const TYPOGRAPHY_RATIO: number
