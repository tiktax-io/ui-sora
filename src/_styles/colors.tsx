// https://www.figma.com/file/RSwdamseVHwjNhSUeiEKsH/Design-Systems-Generator?node-id=418%3A12

import { Color } from '../_types'

/******************************************************************************
 * Browser system palette colors
 *****************************************************************************/
export const aqua: Color = '#00C2D1'
export const black: Color = '#31393C'
export const blue: Color = '#0070ac'
export const green: Color = '#7CEA9C'
export const lime: Color = '#6BF178'
export const navy: Color = '#222E50'
export const orange: Color = '#FA8600'
export const pink: Color = '#FFA9A3'
export const purple: Color = '#C2AFF0'
export const red: Color = '#F45B69'
export const teal: Color = '#028090'
export const yellow: Color = '#F9DC5C'

/******************************************************************************
 * Default theme colors
 *****************************************************************************/
export const primary: Color = blue
export const success: Color = green
export const danger: Color = red
export const text: Color = black
