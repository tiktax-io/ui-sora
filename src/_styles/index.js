'use strict'
/******************************************************************************
 * Below are all global rules applied on theme by default.
 *****************************************************************************/
Object.defineProperty(exports, '__esModule', { value: true })
exports.TYPOGRAPHY_RATIO = exports.TYPOGRAPHY_BASE = exports.BREAKPOINTS = void 0
exports.BREAKPOINTS = [576, 768, 992, 1200, 1400]
exports.TYPOGRAPHY_BASE = 14
exports.TYPOGRAPHY_RATIO = 1.25
