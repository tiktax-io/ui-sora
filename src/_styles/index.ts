/******************************************************************************
 * Below are all global rules applied on theme by default.
 *****************************************************************************/

export const BREAKPOINTS: Array<number> = [576, 768, 992, 1200, 1400]
export const TYPOGRAPHY_BASE: number = 14
export const TYPOGRAPHY_RATIO: number = 1.25
