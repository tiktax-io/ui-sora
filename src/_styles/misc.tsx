const base_spacing = 10
const spacing_ratio = 1.25

/******************************************************************************
 * @type {Array<Number>} Each item is an integer number that represents a
 * lenght in pixels.
 *****************************************************************************/
export const spacing_scale: Array<number> = [
  base_spacing,
  base_spacing * Math.pow(spacing_ratio, 1),
  base_spacing * Math.pow(spacing_ratio, 2),
  base_spacing * Math.pow(spacing_ratio, 3),
  base_spacing * Math.pow(spacing_ratio, 4),
  base_spacing * Math.pow(spacing_ratio, 5)
]

/******************************************************************************
 * @type {Array<Number>} integer number for each breakpoint of the screen
 * width in pixels.
 *****************************************************************************/
export const widthBreakpoints = [
  // Mobile: 360 x 640 or 360 x .
  360,
  // Mobile: 375 x 667 or iPhone X: 375 x 812.
  375,
  // Pixel 2: 411 x 731.
  411,
  // Tablet (portrait): 768 x 1024.
  768,
  // Tablet (landscape): 768 x 1024.
  1024,
  // Laptop: 1366 x 768.
  1366,
  // High-res laptop or desktop: 1920 x 1080.
  1920
]
