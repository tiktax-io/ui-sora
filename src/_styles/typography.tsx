export const typography_ratio: number = 1.25
export const typography_base: number = 14
export const fontFamily: string = '"Inter", sans-serif'

/******************************************************************************
 * Autogenerated styles. Only modify above this line.
 *****************************************************************************/

export const typography_scale: Array<number> = [
  typography_base * Math.pow((1/typography_ratio), 3),
  typography_base * Math.pow((1/typography_ratio), 2),
  typography_base * Math.pow((1/typography_ratio), 1),
  typography_base,
  typography_base * Math.pow(typography_ratio, 1),
  typography_base * Math.pow(typography_ratio, 2),
  typography_base * Math.pow(typography_ratio, 3),
  typography_base * Math.pow(typography_ratio, 4),
  typography_base * Math.pow(typography_ratio, 5)
]
