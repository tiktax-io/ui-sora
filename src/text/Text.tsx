import React, { createElement, DetailedReactHTMLElement, forwardRef, HTMLAttributes, LegacyRef, memo, ReactElement } from 'react'
import { Color } from '../_types'
import { CSSinJS, TextSize, TextType } from '../_types/global'

import { useStyles } from './Text.styles'
import { TextProps } from './Text.types'

/******************************************************************************
 * @param {string} children String containing the text to be rendered.
 * @param {Color} color Specify color of text.
 * @param {CSSinJS} css Css properties to add.
 * @param {TextProps['element']} element HTML element that will wrap text.
 * @param {TextSize} size Natural number (can be negative) representing scale of
 * the font. 0 is the base size, -1 is one size smaller than base, 1 is one
 * size greater than base. If left blank size will be 0.
 * @param {CSSinJS['fontWeight']} weight. Specify weight from 100 to 900.
 * @returns {ReactElement<TextType>} Text element that will depend on props.
 * "h1"-"h6" headings, "p" paragraphs or span elements.
 *****************************************************************************/
const Text = memo(forwardRef((props: TextProps, ref: LegacyRef<TextType>):
  DetailedReactHTMLElement<HTMLAttributes<HTMLElement>, HTMLElement> => {
  const {
    children,
    color,
    css,
    element,
    size = 0,
    weight = 'normal'
  } = props
  const classes = useStyles()

  /****************************************************************************
   * Gets the Text component prop "size" representing the size of the text as
   * an integer number and returns a string representing an HTML Heading
   * element to use with "createElement" method of React API.
   * @returns {'h1' | 'h2' | 'h3' | 'h4'} heading HTML Element
   ***************************************************************************/
  const getHeadingElement = (size: TextSize) => {
    switch(size) {
      case 5:
        return 'h1'
      case 4:
        return 'h2'
      case 3:
        return 'h3'
      case 2:
        return 'h4'
      default:
        return 'h4'
    }
  }

  /****************************************************************************
   * Gets the Text component prop "element" and inferes what type of HTML
   * elemnt to render.
   * @returns a string representing the HTML element to use with
   * "createElement" method of React API.
   ***************************************************************************/
  const getElement = (element: TextProps['element']) =>
    element === 'span'
      ? 'span'
      : element === 'paragraph'
        ? 'p'
        : size > 1
          ? getHeadingElement(size)
          : 'span'

  const attributes = {
    ...props,
    className: classes.default,
    ref: ref,
    style: {
      ...css,
      color: color,
      fontWeight: weight
    }
  }
  const htmlElement = getElement(element)
  return createElement(htmlElement, attributes, children)
}))

export default Text
