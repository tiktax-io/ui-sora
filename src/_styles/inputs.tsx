import { hexToRGB, shadeColor } from '../_utils/colors'
import { primary } from './colors'
import { spacing_scale } from './misc'
import { fontFamily, typography_base, typography_scale } from './typography'

export const inputGlobalStyles = {
  container: {
    border: 'none',
    borderRadius: '10px',
    height: '40px',
    fontFamily: fontFamily,
    fontSize: typography_base,
    paddingLeft: `${spacing_scale[3]}px`,
    paddingRight: `${spacing_scale[3]}px`,
    '& span': {
      fontSize: typography_scale[3],
    },
    '&:disabled': {
      background: shadeColor('#ffffff', 1.05),
      border: '1px solid transparent',
      boxShadow: 'none',
      cursor: 'not-allowed',
      '& span': {
        color: shadeColor('#ffffff', 1.14)
      },
      '&:active': {
        background: shadeColor('#ffffff', 1.05),
        boxShadow: 'none'
      },
      '&:focus': {
        background: shadeColor('#ffffff', 1.05),
        boxShadow: 'none'
      },
      '&:hover': {
        background: shadeColor('#ffffff', 1.05),
        boxShadow: 'none'
      }
    },
    '&:focus-within': {
      border: `1px solid ${hexToRGB(primary, 0.5)}`,
      boxShadow: `0px 0px 4px 5px ${hexToRGB(primary, 0.15)}`,
      outline: 'none'
    }
  }
}
