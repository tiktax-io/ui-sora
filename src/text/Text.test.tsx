import React from 'react'
import { render, screen } from '@testing-library/react'
import Text from './Text'
import renderer from 'react-test-renderer'
import { danger } from '../_styles'

describe('Text', () => {

  test('Renders text as children', () => {
    render(<Text>Hola</Text>)
    const demoElement = screen.getByText(/Hola/i)
    expect(demoElement).toBeInTheDocument()
  })

  test('Can customize color via props', () => {
    const tree = renderer
      .create(
        <Text color={danger}>My custom text</Text>
      )
      .toJSON()
    
    expect(tree).toMatchSnapshot()
  })

  test('Can specify weight via props', () => {
    const tree = renderer
      .create(
        <Text weight={700}>My custom text</Text>
      )
      .toJSON()
    
    expect(tree).toMatchSnapshot()
  })

})
