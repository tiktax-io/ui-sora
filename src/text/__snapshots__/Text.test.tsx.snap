// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Can customize color via props 1`] = `
<span
  className="default-0-2-1"
  color="#F45B69"
  style={
    Object {
      "color": "#F45B69",
      "fontWeight": "normal",
    }
  }
>
  My custom text
</span>
`;

exports[`Text Can specify weight via props 1`] = `
<span
  className="default-0-2-1"
  style={
    Object {
      "color": undefined,
      "fontWeight": 700,
    }
  }
  weight={700}
>
  My custom text
</span>
`;
